Add support and doc for AD579x DAC

SaikiranGudla opened the pull request:
Add precision toolbox support for AD5760, AD5780,
AD5781, AD5790, AD5791 DACs

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/analogdevicesinc/PrecisionToolbox?pullRequest=14) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/analogdevicesinc/PrecisionToolbox?pullRequest=14) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/analogdevicesinc/PrecisionToolbox?pullRequest=14) it.</sub>

tfcollins commented on +adi/+AD5760/Tx.m:
```diff
@@ -0,0 +1,133 @@
+    %   <a href="https://www.analog.com/media/en/technical-documentation/data-sheets/ad5760.pdf">AD5760 Datasheet</a>
+
+    properties 
+        SampleRate
+        CodeSelect
+        PowerDown
+        Raw
```
All properties should have a default value with docstrings

tfcollins commented on +adi/+AD5760/Tx.m:
```diff
@@ -0,0 +1,133 @@
+            % Set device sampling rate
+            obj.SampleRate = value;
+            if obj.ConnectedToDevice
+                obj.setDeviceAttributeRAW('sampling_frequency', num2str(value));            end
```
Can you move the "end" to the next line

tfcollins commented on +adi/+AD5760/Tx.m:
```diff
@@ -0,0 +1,133 @@
+    properties 
+        SampleRate
+        CodeSelect
+        PowerDown
```
This should be a Logical

tfcollins commented on +adi/+AD5760/Tx.m:
```diff
@@ -0,0 +1,133 @@
+        end
+
+        function set.PowerDown(obj, value)
+            options = [0,1];
+            if ~any(value==options)
+               error(['PowerDown must be one of ',num2str(options)]);
+            end
```
If you define PowerDown as a Logical this check is done for you.

tfcollins reviewed (changes requested):
All the child classes are almost identical except for the device names. You should instead put all the properties in the base class an update the devName+phyDevName in the constructor calls. This greatly simplifies the code

ribdp commented on +adi/+AD5781/Tx.m:
```diff
@@ -0,0 +1,134 @@
+            end
+        end
+
+                function set.CodeSelect(obj, value)
```
Indentation is a bit off

SaikiranGudla commented:
> All the child classes are almost identical except for the device names. You should instead put all the properties in the base class an update the devName+phyDevName in the constructor calls. This greatly simplifies the code

Yes, except for the device names, and resolution (data type), all the generics are almost identical. Do you suggest using a single Tx class for all the generics with the device name as an argument?

SaikiranGudla commented:
v1.0: Addressed review comments

1. Moved all the properties to the base class and updated the devName, phyDevName, and dataType in the constructor calls
2. Defined PowerDown as logical
3. Added comments for properties

tfcollins commented on +adi/+AD579x/Base.m:
```diff
@@ -0,0 +1,155 @@
+        % SamplesPerFrame Samples Per Frame
+        %   Number of samples per frame, specified as an even positive
+        %   integer.        
+         SamplesPerFrame
```
Please provide a default value

tfcollins reviewed (changes requested):
Just update the SamplesPerFrame and this is ready to merge

SaikiranGudla commented:
v 1.1: Addressed review comments

Added default value for SamplesPerFrame
